dao-contracts/src/voting/governance_voting.rs

```diff
@@ -0,0 +1,366 @@
+//! Governance Voting module.
+pub mod consts;
+pub mod events;
+pub mod voting;
+
+use casper_dao_utils::bytes::BytesConversion;
+
+use casper_dao_utils::{
+    casper_contract::unwrap_or_revert::UnwrapOrRevert,
+    casper_dao_macros::Instance,
+    casper_env::{call_contract, emit, get_block_time, revert, self_address},
+    Address, Error, Mapping, Variable,
+};
+use casper_types::{runtime_args, RuntimeArgs, U256, U512};
+
+use crate::{
+    ReputationContractCaller, ReputationContractInterface, VariableRepositoryContractCaller,
+};
+
+use self::{
+    events::{
+        FormalVotingEnded, InformalVotingEnded, VoteCast, VotingContractCreated, VotingCreated,
+    },
+    voting::{Voting, VotingConfiguration, VotingResult, VotingType},
+};
+
+use casper_dao_utils::consts as dao_consts;
+
+use super::{vote::VotingId, Vote};
+
+/// The Governance Voting module.
+
+#[derive(Instance)]
+pub struct GovernanceVoting {
+    pub variable_repo: Variable<Option<Address>>,
+    pub reputation_token: Variable<Option<Address>>,
+    pub votings: Mapping<U256, Voting>,
+    pub votes: Mapping<(U256, Address), Vote>,
+    pub voters: Mapping<U256, Vec<Address>>,
+    pub votings_count: Variable<U256>,
+    pub dust_amount: Variable<U256>,
+}
+
+impl GovernanceVoting {
+    /// Initialize the module.
+    pub fn init(&mut self, variable_repo: Address, reputation_token: Address) {
+        self.variable_repo.set(Some(variable_repo));
+        self.reputation_token.set(Some(reputation_token));
+
+        emit(VotingContractCreated {
+            variable_repo,
+            reputation_token,
+            voter_contract: self_address(),
+        });
+    }
+
+    pub fn create_voting(
+        &mut self,
+        creator: Address,
+        stake: U256,
+        contract_to_call: Address,
+        entry_point: String,
+        runtime_args: RuntimeArgs,
+    ) {
+        let repo_caller = VariableRepositoryContractCaller::at(self.get_variable_repo_address());
+
+        let reputation_caller = ReputationContractCaller::at(self.get_reputation_token_address());
+
+        let informal_voting_time = repo_caller.get_variable(dao_consts::INFORMAL_VOTING_TIME);
+        let formal_voting_time = repo_caller.get_variable(dao_consts::FORMAL_VOTING_TIME);
+        let minimum_governance_reputation =
+            repo_caller.get_variable(dao_consts::MINIMUM_GOVERNANCE_REPUTATION);
+        let voting_id = self.votings_count.get();
+
+        let informal_voting_quorum = VariableRepositoryContractCaller::mul_by_ratio(
+            reputation_caller.total_onboarded(),
+            repo_caller.get_variable(dao_consts::INFORMAL_VOTING_QUORUM),
+        );
+        let formal_voting_quorum = VariableRepositoryContractCaller::mul_by_ratio(
```

Comment:
`mul_by_ratio` definitely should not be a part of `VariableRepositoryContractCaller`. Put it in some utils as a standalone function.